frontend/src/components/ingress/Details.tsx

```diff
@@ -42,6 +43,113 @@ export default function IngressDetails() {
     );
   }
 
+  /** this function is used to determine the protocol used in the formatURL for LinkedStringFormat */
+  function isHttpsUsed(item: Ingress, url: String) {
+    const hostList: string[] = item.jsonData.spec?.tls?.map(({ ...hosts }) => `${hosts.hosts}`);
+    const isHttps = hostList.includes(`${url}`);
+
+    return isHttps;
+  }
+
+  /** this function is used to format the url to be used in the Link component */
+  function LinkStringFormat(url: String, item: Ingress, urlPath?: String) {
```

Comment:
How about adding a story for LinkStringFormat? You could add the different states, which would be a nice test.

Reply:
added!